include/rocksdb/cache.h

```diff
@@ -532,8 +533,23 @@ class Cache {
   // to each of the handles.
   virtual void WaitAll(std::vector<Handle*>& /*handles*/) {}
 
+  // Reserves (block) cache space for memory used in this cache.
+  void SetCacheReservationManager(
+      std::shared_ptr<ConcurrentCacheReservationManager> cache_res_mgr) {
+    if (cache_res_mgr_ != cache_res_mgr) {
+      cache_res_mgr_ = cache_res_mgr;
+    }
+  }
+
+  ConcurrentCacheReservationManager* cache_reservation_manager() const {
+    return cache_res_mgr_.get();
+  }
+
  private:
   std::shared_ptr<MemoryAllocator> memory_allocator_;
+
+  // ONLY USED for charging blob cache usage
+  std::shared_ptr<ConcurrentCacheReservationManager> cache_res_mgr_ = nullptr;
```

Comment:
Probably we should move `SetCacheReservationManager()` and `std::shared_ptr<ConcurrentCacheReservationManager> cache_res_mgr_ = nullptr;` to `sharded_cache.h`. We don't want to provide users access to any special APIs.